### DISCUSSION TITLE
Any update on the text field feature?

### DISCUSSION THREAD

**Discussion by VarLad** [flxzt/rnote#33] [Development]
Any idea by when we'll have text field feature in rnote?
Both markdown and plain (with preferably syntax highlighting in Markdown)

**Comment by flxzt** [comment]
Can't tell you tbh. I think text is not so straight forward to implement and Rnote is a hobby project. And while it is on my mind so are many other features like sharing via libp2p, tabs, etc.

**Reply by VarLad** [reply]
@flxzt Thanks for replying! 
The project is darn awesome anyway, but text fields in someway would be nice :)
while I'm not a GTK4 dev, maybe https://developer.gnome.org/gtk3/stable/TextWidget.html and https://wiki.gnome.org/Projects/GtkSourceView might help in a straightforward implementation? 
Again, I've no idea if or how it'll work

Again, thanks for the awesome project! I'm already using it for most of my study stuff :smile:

**Reply by flxzt** [reply]
The text basically just needs to be able to be generated as a SVG to be renderable. I think either Rnote could draw a a existing text widget on top of the canvas while writing, but that would require careful attention that it generates to a svg exactly the same, or as a custom text widget where everything would need to be implemented, which I think is better, because we could do it exactly as we want, but is a lot harder and is a lot more effort.
